Proxify tests failing

There are currently 3 proxify tests that are failing in CI:

```python
FAILED dask_cuda/tests/test_proxify_host_file.py::test_dataframes_share_dev_mem - AttributeError: 'types.SimpleNamespace' object has no attribute '_owner'
FAILED dask_cuda/tests/test_proxify_host_file.py::test_incompatible_types - assert array([0, 1, 2, 3, 4, 5, 6, 7, 8]) is array([0, 1, 2, 3, 4, 5, 6, 7, 8])
FAILED dask_cuda/tests/test_proxify_host_file.py::test_worker_force_spill_to_disk - AssertionError: assert 1379766272 < (1379766272 - (10 ** 7))
```

I haven't looked at those in detail, but an environment I had from early in the day doesn't fail, whereas a current one does. Given that, my primary suspect would be https://github.com/rapidsai/cudf/pull/12009 , any ideas @madsbk ?

--- a/dask_cuda/get_device_memory_objects.py
+++ b/dask_cuda/get_device_memory_objects.py
@@ -51,8 +51,9 @@ def get_device_memory_objects_default(obj):
         return dispatch(obj._pxy_get().obj)
     if hasattr(obj, "data"):
         return dispatch(obj.data)
-    if hasattr(obj, "_owner") and obj._owner is not None:
-        return dispatch(obj._owner)
+    owner = getattr(obj, "owner", None) or getattr(obj, "_owner", None)
+    if owner:
+        return dispatch(owner)
     if hasattr(obj, "__cuda_array_interface__"):
         return [obj]
     return []

--- a/dask_cuda/tests/test_proxify_host_file.py
+++ b/dask_cuda/tests/test_proxify_host_file.py
@@ -273,7 +273,7 @@ def test_dataframes_share_dev_mem():
     # Even though the two dataframe doesn't point to the same cudf.Buffer object
     assert view1["a"].data is not view2["a"].data
     # They still share the same underlying device memory
-    assert view1["a"].data._owner._owner is view2["a"].data._owner._owner
+    view1["a"].data.ptr == view2["a"].data.ptr
 
     dhf = ProxifyHostFile(
         local_directory=root_dir, device_memory_limit=160, memory_limit=1000

